client/rest-high-level/src/test/java/org/elasticsearch/client/BulkProcessorIT.java

```diff
@@ -448,33 +448,41 @@ private static MultiGetRequest indexDocs(BulkProcessor processor, int numDocs, S
             } else {
                 BytesArray data = bytesBulkRequest(localIndex, localType, i);
                 processor.add(data, globalIndex, globalType, globalPipeline, null, XContentType.JSON);
+
+                if (localType != null) {
```

Comment:
This line troubled me. Bulk requests can pass a type on each request line (`localType` var here) or pass a default fallback choice which is inherited by any request lines that don't declare a localType (the fallback is called `globalType`).
So here, I assumed the test's logic should ideally be:
 
     if (localType !=null || globalType != null) {

These are the conditions under which a user has sent a choice of custom type to the server.
Sadly this doesn't work. It looks like in the server-side `BulkRequest` that only types expressed at the line level are flagged. It assumes any "global" references to type are already flagged by the main RestBulkAction which we are not exercising in this test. What we are squashing with these `assertWarnings` checks in BulkProcessorIT are the ThreadContext values normally set up on the server side by the server-side objects like BulkRequest which we are (questionably) re-using in HLRC. 
These warning checks are not really the same as simply checking that the response from a _bulk REST interaction has the expected warning if you use happen to use global or local type references in the request. That contract is exercised in `BulkRequestWithGlobalParametersIT` so I think we're good there.